packages/help-center/src/hooks/use-zendesk-config.ts

```diff
@@ -0,0 +1,18 @@
+import { useQuery } from '@tanstack/react-query';
+
+async function requestZendeskConfig() {
+	return await window.fetch( 'https://wpcom.zendesk.com/embeddable/config', {
+		method: 'GET',
+	} );
+}
+
+export default function useZendeskConfig( enabled: boolean ) {
+	return useQuery< Response >( [ 'getZendeskConfig' ], requestZendeskConfig, {
+		staleTime: Infinity,
+		retry: false,
+		meta: {
+			persist: false,
+		},
+		enabled,
```

Comment:
You can add a `select` here to process the resolved data to add another field for 3rd party cookies, then you can do:

```js
const { thirdPartyCookieAllowed } = useZendeskConfig( Boolean( chatStatus?.is_user_eligible ) );
```

...anywhere you want without all the store boilerplate, the effects, and you get free caching.

Reply:
I like the idea! Definitely happy with less boilerplate. Unfortunately, `select` does not seem to work my use-case. From my testing, it seems it's only called when the promise resolves. If there's a network error (which is exactly what I'm interested in here), it's never called.
But this gave me an idea to simplify this code and indeed just use `useZendeskConfig` directly.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/76777/commits/f4b09067cd81bba7d2c79efae4a40c6189c537d2.
While testing I notice that despite having `retry: false`, react-query _really_ seems to hate the request failing and tries to retry it on focus, mount, etc. So I had to explicitly disable those too.